src/cmap/test/integration.rs

```diff
@@ -122,20 +122,22 @@ async fn concurrent_connections() {
     });
     futures::future::join_all(tasks).await;
 
-    // ensure all three ConnectionCreatedEvents were emitted before one ConnectionReadyEvent.
-    let events = handler.events.read().unwrap();
-    let mut consecutive_creations = 0;
-    for event in events.iter() {
-        match event {
-            Event::ConnectionCreated(_) => {
-                consecutive_creations += 1;
-            }
-            Event::ConnectionReady(_) => {
-                if consecutive_creations < 2 {
-                    panic!("connections not created concurrently");
+    {
```

Comment:
I tried using `drop(events)` on line 141 instead of using a block, but clippy strangely still gave the error about `events` being held across an `await` point

Reply:
Yeah this appears to be a bug in clippy: https://github.com/rust-lang/rust-clippy/issues/6446

Reply:
Ah, nice catch